scripts/main.ts

```diff
@@ -0,0 +1,128 @@
+        }
+
+        if (configuration.serviceEndpointUIExtensionDetails.data.scope) {
+            viewModel.scope(configuration.serviceEndpointUIExtensionDetails.data.scope);
+        }
+
+        if (configuration.serviceEndpointUIExtensionDetails.data.audience) {
+            viewModel.audience(configuration.serviceEndpointUIExtensionDetails.data.audience);
+        }
+
+        if (configuration.serviceEndpointUIExtensionDetails.data.issuer) {
+            viewModel.issuer(configuration.serviceEndpointUIExtensionDetails.data.issuer);
+        }
+
+        if (configuration.serviceEndpointUIExtensionDetails.data.projectid) {
+            viewModel.projectid(configuration.serviceEndpointUIExtensionDetails.data.projectid);
+        }
+
+        if (configuration.serviceEndpointUIExtensionDetails.data.privatekey) {
+            viewModel.privatekey(configuration.serviceEndpointUIExtensionDetails.data.privatekey);
+        }
+    }
+
+    configuration.validateEndpointDetailsFuncImpl(function () {
+        if ((!viewModel.connectionName()) || (!viewModel.scope()) || (!viewModel.certificate())) {
+            viewModel.errors("All fields are required.");
+            return false;
+        }
+
+        try {
+            var jsonKeyFileContent = JSON.parse(viewModel.certificate());
+        } catch (e) {
+            viewModel.errors("Please provide valid json in 'JSON Key File' field");
+            return false;
+        }
+
+        if ((!jsonKeyFileContent.client_email) || (!jsonKeyFileContent.token_uri) || (!jsonKeyFileContent.private_key) || (!jsonKeyFileContent.project_id)) {
+            viewModel.errors("All fields are required.");
+            return false;
+        }
+
+        return true;
+    });
+
+    configuration.getEndpointDetailsFuncImpl(function () {
+
+        var serviceEndpoint: any = {};
+        var gcpEndpointDetails: any = {};
+        var jsonKeyFileContent = JSON.parse(viewModel.certificate());
+        gcpEndpointDetails.issuer = jsonKeyFileContent.client_email;
+        gcpEndpointDetails.audience = jsonKeyFileContent.token_uri;
+        gcpEndpointDetails.privatekey = jsonKeyFileContent.private_key;
+        gcpEndpointDetails.projectid = jsonKeyFileContent.project_id;
+
+        serviceEndpoint.data = {
+            projectid: gcpEndpointDetails.projectid
+        };
+        serviceEndpoint.authorization = {
+            parameters: {
+                certificate: viewModel.certificate(),
+                scope: viewModel.scope(),
+                issuer: gcpEndpointDetails.issuer,
+                audience: gcpEndpointDetails.audience,
+                privatekey: gcpEndpointDetails.privatekey
+            },
+            scheme: "JWT"
+        };
+        serviceEndpoint.url = "https://www.googleapis.com/";
+        serviceEndpoint.name = viewModel.connectionName();
+        var serviceEndpointUIExtensionDetails: ServiceEndpointUIExtensionDetails = serviceEndpoint;
```

Comment:
Use `as` to assert a variable has a particular type.

Reply:
Fixed it in next iteration.